Assets/Scripts/Player/PhysicsHands/GrabLogic.cs

```diff
@@ -0,0 +1,144 @@
+using UnityEngine;
+using System.Collections.Generic;
+using Unity.XR.CoreUtils.Datums;
+using System;
+
+namespace RogueApeStudios.SecretsOfIgnacios
+{
+    public class GrabLogic : MonoBehaviour
+    {
+        private Rigidbody _grabbedRb;
+        private int _oldGrabbableLayer;
+        private LayerMask _grabLayerMask;
+        private float _initialGrabDist;
+        private ConfigurableJoint _grabbedJoint;
+        [SerializeField] private float _grabRange;
+        [SerializeField] private Rigidbody _grabbingRb;
+        [SerializeField] private List<String> _layersToIgnore;
+        [SerializeField] private Transform _grabCenter;
+        // Start is called once before the first execution of Update after the MonoBehaviour is created
+        void Start()
+        {
+            LayerMask mask = Physics.AllLayers;
+
+            foreach (string layername in _layersToIgnore)
+            {
+                LayerMask layer = LayerMask.NameToLayer(layername);
+                mask &= ~(1 << layer);
+            }
+            _grabLayerMask = mask;
+        }
+
+        // Update is called once per frame
+        void FixedUpdate()
+        {
+            //check if the grabbed object is close enough, and end grab if it is too far away
+            if (_grabbedRb != null) 
+            {
+                var dist = Vector3.Magnitude(_grabbedRb.position - _grabCenter.position) - _initialGrabDist;
+                if(dist > .5 * _grabRange)
+                {
+                    EndGrab();
+                }
+            }
+
+        }
+
+
+        public void CheckForGrabbables()
+        {
+            Debug.Log("i'm gonna grab u!!!!1");
```

Comment:
Try to remove logs if they don't provide specific info, this should not be necessary in dev after approval